src/ioctls/vm.rs

```diff
@@ -1394,6 +1394,50 @@ impl VmFd {
         self.check_extension_int(c) > 0
     }
 
+    /// Creates an anonymous file and returns a file descriptor that refers to it.
+    ///
+    /// See the documentation for `KVM_CREATE_GUEST_MEMFD`.
+    ///
+    /// Returns an io::Error when the file could not be created.
+    ///
+    /// # Arguments
+    ///
+    /// * kvm_create_guest_memfd - KVM create guest memfd structure. For details check the
+    ///                    `kvm_create_guest_memfd` structure in the
+    ///                    [KVM API doc](https://www.kernel.org/doc/Documentation/virtual/kvm/api.txt).
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # extern crate kvm_ioctls;
+    /// extern crate kvm_bindings;
+    ///
+    /// # use kvm_ioctls::Kvm;
+    /// use kvm_bindings::kvm_create_guest_memfd;
+    ///
+    /// let kvm = Kvm::new().unwrap();
+    /// let vm = kvm.create_vm().unwrap();
+    /// let gmem = kvm_create_guest_memfd{
+    ///     size: 0x1000,
+    ///     flags: 0,
+    ///     reserved: [0; 6],
+    /// };
+    ///
+    /// let id: __u32 = unsafe {
+    ///     vm_fd.create_guest_memfd(gmem).unwrap()
+    /// };
+    /// ```
+    pub unsafe fn create_guest_memfd(
+        &self,
+        gmem: kvm_create_guest_memfd,
+        ) -> Result<__u32> {
```

Comment:
Maybe this could return RawFd, which is effectively i32 but more expressive

Reply:
I think I used u32 here due to the `kvm_userspace_memory_region2` structure:
```
struct kvm_userspace_memory_region2 {
	__u32 slot;
	__u32 flags;
	__u64 guest_phys_addr;
	__u64 memory_size;
	__u64 userspace_addr;
	__u64 guest_memfd_offset;
	__u32 guest_memfd; // this supposes to contain the file created via KVM_CREATE_GUEST_MEMFD
	__u32 pad1;
	__u64 pad2[14];
};
```